File: dotnet-procapi/Preprocessors/OutputPreprocessor.cs
using System.Collections.Generic;
using ProcAPI.Contracts;

namespace ProcAPI.Preprocessors
{
    public class OutputPreprocessor : IPreprocessor<string, OutputData>
    {
        private readonly Dictionary<string, string> _map;

        public OutputPreprocessor()
        {
            _map = new Dictionary<string, string>
            {
                {"0", "cat"},
                {"1", "dog"}
            };
        }

        public OutputData PreprocessData(string data) => new OutputData
        {
            ResultClass =
                _map.TryGetValue(data, out var value)
                    ? value
                    : "unknown"
        };
    }
}

File: python-procapi/Preprocessors/OutputPreprocessor.py
from Contracts.OutputData import OutputData
from Preprocessors.IPreprocessor import IPreprocessor


class OutputPreprocessor(IPreprocessor):
    __result_map = {
        "0": "cat",
        "1": "dog"
    }

    def preprocess_data(self, data: str) -> OutputData:
        return OutputData(result_class=self.__result_map.get(data, "unknown"))
